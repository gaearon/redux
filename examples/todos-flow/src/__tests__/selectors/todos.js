// @flow

import { visibleTodosSelector } from '../../selectors';

describe('visibleTodosSelector', () => {
  let state;
  beforeEach(() => {
    state = {
      todos: [
        { id: 0, text: 'Test Todo', completed: false },
        { id: 0, text: 'Test Todo', completed: false }
      ],
      visibilityFilter: 'SHOW_ALL'
    };
  });

  test('should handle SHOW_ALL', () => {
    expect(visibleTodosSelector(state)).toMatchSnapshot();
  });

  test('should handle SHOW_ACTIVE', () => {
    state.visibilityFilter = 'SHOW_ACTIVE';
    expect(visibleTodosSelector(state)).toMatchSnapshot();
  });

  test('should handle SHOW_COMPLETED', () => {
    state.visibilityFilter = 'SHOW_COMPLETED';
    expect(visibleTodosSelector(state)).toMatchSnapshot();
  });
});
