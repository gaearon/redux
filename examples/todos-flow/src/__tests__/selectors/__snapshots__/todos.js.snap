exports[`visibleTodosSelector should handle SHOW_ACTIVE 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Test Todo",
  },
  Object {
    "completed": false,
    "id": 0,
    "text": "Test Todo",
  },
]
`;

exports[`visibleTodosSelector should handle SHOW_ALL 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Test Todo",
  },
  Object {
    "completed": false,
    "id": 0,
    "text": "Test Todo",
  },
]
`;

exports[`visibleTodosSelector should handle SHOW_COMPLETED 1`] = `Array []`;
